### PR TITLE
build: add CI workflow 🕺

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
- [x] Enable CI workflow
- [x] Enable Pipeline workflow
- [x] Fix event on which CT workflow runs

Ref PR: #597

Signed-off-by: Vinayak Kulkarni <19776877+vinayakkulkarni@users.noreply.github.com>

### CODE REVIEW

**Comment by acalcutt** [comment]
I tried to test this, but I keep getting stuck on the linting part of the ci. Any idea what may be wrong? If I run 'npm run lint:eslint' I am only getting warnings, and lint:prettier says it matches the code style.

https://github.com/acalcutt/tileserver-gl/actions/runs/3860934983/jobs/6581588926
image

**Comment by vinayakkulkarni** [comment]
> I tried to test this, but I keep getting stuck on the linting part of the ci. Any idea what may be wrong? If I run 'npm run lint:eslint' I am only getting warnings, and lint:prettier says it matches the code style.
> 
> [acalcutt/tileserver-gl/actions/runs/3860934983/jobs/6581588926](https://github.com/acalcutt/tileserver-gl/actions/runs/3860934983/jobs/6581588926) image

Annotations for ESLint – https://github.com/acalcutt/tileserver-gl/runs/10497991087
Annotations for Prettier – https://github.com/acalcutt/tileserver-gl/runs/10497991095

Fixed the eslint & prettier issues – [maptiler/tileserver-gl@`69acad2` (#671)](https://github.com/maptiler/tileserver-gl/pull/671/commits/69acad272b91ee63ff427a8e6ec5609849ca0d9f)

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    uses: maptiler/tileserver-gl/.github/workflows/ct.yml@master
+  automerger:
+    name: 'Automerge Dependabot PRs'
+    needs:
+      - ci
+      - ct
+      - extract-branch
+    if: >
+      github.event_name == 'pull_request' && github.event.pull_request.user.login == 'dependabot[bot]'
+    uses: maptiler/tileserver-gl/.github/workflows/ct.yml@master
```
Comment (acalcutt) [inline .github/workflows/pipeline.yml:43]: this should point to automerger.yml

Comment (vinayakkulkarni) [reply .github/workflows/pipeline.yml]: - [x] Done in ab78ce2

**Comment by acalcutt** [comment]
I got past the lint issue with the changes you made, and then the automerge worked.

But then I merged in the new docker build test I added in the last PR and it fails to build the arm64 version. it doesn't seem to like the versions set in the dockerfile. that was one of my concerns with setting the version there.

https://github.com/acalcutt/tileserver-gl/actions/runs/3871285491/jobs/6598947260
image

**Comment by vinayakkulkarni** [comment]
> I got past the lint issue with the changes you made, and then the automerge worked. [acalcutt/tileserver-gl/actions/runs/3871247730](https://github.com/acalcutt/tileserver-gl/actions/runs/3871247730)
> 
> But then I merged in the new docker build test I added in the last PR and it fails to build the arm64 version. it doesn't seem to like the versions set in the dockerfile. that was one of my concerns with setting the version there.
> 
> [acalcutt/tileserver-gl/actions/runs/3871285491/jobs/6598947260](https://github.com/acalcutt/tileserver-gl/actions/runs/3871285491/jobs/6598947260) image

Probably because `arm64` has different package version and `amd` has different

Screenshot 2023-01-10 at 12 00 31 AM

xvfb – https://packages.ubuntu.com/focal/xvfb
libcurl4 – https://packages.ubuntu.com/focal/libcurl4

**Comment by acalcutt** [comment]
Can we continue to not specify versions in the docker file like we do now?

**Comment by vinayakkulkarni** [comment]
> Can we continue to not specify versions in the docker file like we do now?

- [x] Done in fa779fc

**Review by acalcutt — APPROVED** [review]
My one concern is we are starting to get a lot of random config files in the root directory. not sure if we can do better with that though...just a thought for the future.

**Comment by vinayakkulkarni** [comment]
> My one concern is we are starting to get a lot of random config files in the root directory. not sure if we can do better with that though...just a something to think about for the future.

These files only improve the developer experience and are critical for ensuring multiple things ranging from code standardization, DX, vulnerability scanning etc. 
Those files in any ways don't interfere with the "how" & "why" of tileserver-gl usage.

**Pull request merged**